src/book/mod.rs

```diff
@@ -311,6 +328,10 @@ impl MDBook {
         if failed {
             bail!("One or more tests failed");
         }
+
+        if chapter.is_some() && !chapter_found {
+            bail!(format!("Chapter not found: {:?}", chapter));
+        }
```

Comment:
This produces some strange output. Generally it is recommended to not use debug displays for user output. Also, the `bail!` macro can take formatted strings directly, there is no need to use `format!`. I think it might be better to extract the name through a pattern and use a display impl. Something like this:

```suggestion
        if let Some(chapter) = chapter {
            if !chapter_found {
                bail!("Chapter not found: {}", chapter);
            }
        }
```

Reply:
Much better thanks, I merged your fix.  I was just copying the use of `bail` from a few lines earlier: `bail!("One or more tests failed");`, so unless there is a better error formatter available, I'll leave the bail as you have written it.